specs/plugin-extensibility.md

```diff
@@ -394,7 +411,154 @@ All response attributes are required.
+  "trustPolicy" : {
+    // Array of trusted identities as specified in 
+    // trust policy.
+    "trustedIdentities": [],
+    // Array of verification checks to be performed by the plugin
+    "signatureVerification" : [ 
+        "SIGNATURE_VERIFIER.TRUSTED_IDENTITY",
+        "SIGNATURE_VERIFIER.REVOCATION_CHECK"
+        ]
+  },
+  
+  // Optional plugin configuration, map of string-string
+  "pluginConfig" : { }
+}
+```
+
+The request can be extended in future by including other elements of signature envelope and trust policy as required to customize additional steps in verification.
+
+*Response*
+All response attributes are required.
+
+```jsonc
+{
+  "verificationResults" : {
+    // Map of results. Key must match the set of
+    // verification capabilities 
+    // in verify-signature request's trustPolicy.signatureVerification attribute.
+    //  SIGNATURE_VERIFIER.TRUSTED_IDENTITY
+    //  SIGNATURE_VERIFIER.REVOCATION_CHECK
+        "SIGNATURE_VERIFIER.TRUSTED_IDENTITY" :
+        {
+            // The result of a verification check
+            "success" : true | false,
+            // Reason for check being successful or not,
+            // required if value of success attribute is false.
+            "reason" : ""
+        },
+        "SIGNATURE_VERIFIER.REVOCATION_CHECK" :
+        {
+            // The result of a verification check
+            "success" : true | false,
+            // Reason for check being successful or not,
+            // required if value of success attribute is false.
+            "reason" : ""
+        }
+  },
+  // Array of strings containing critical attributes processed by the plugin.
+  "processedAttributes" : [ 
+  ]
+}
+```
+
+*verificationResults* : Verifications performed by the plugin. This is a map where the keys MUST match set of verify capabilities in `verify-signature` request's `trustPolicy.signatureVerification` attribute, and values are objects with following attributes
+
+* *success* (required): The `boolean` verification result.
+* *reason* (optional): Reason associated with verification being successful or not, REQUIRED if value of success field is `false`.
+
+*processedAttributes* (required): Array of strings containing critical attributes processed by the plugin. Values must be one or more of attribute names in `verify-signature` request's `signature.unprocessedAttributes`.
+
+#### Error codes for *verify-signature*
+
+* VALIDATION_ERROR - Any of the required request fields was empty, or a value was malformed/invalid.
+* UNSUPPORTED_CONTRACT_VERSION - The contract version used in the request is unsupported.
+* ACCESS_DENIED - Authentication/authorization during signature verification, this may be due to external calls made by the plugin.
+* TIMEOUT - The verification process timed out and can be retried by Notation.
+* THROTTLED - The verification process was throttled and can be retried by Notation.
+* ERROR - Any general error that does not fall into previous error categories.
```

Comment:
Which field should we put the error code?

Reply:
Errors are returned as a separate error response JSON with non zero exit code - https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#plugin-contract